### PR TITLE
Add missing ghost modifier when printing VarDeclStmt

### LINKED ISSUE #1280: Pretty-printer does not preserve `ghost var`
When pretty-printing the following erroneous program, dafny doesn't print the "ghost" keyword. As a result, the pretty-printed program verifies.

```
method Test() returns (r: int)
{
    ghost var (x, y) := (123, 234);
    r := x; // expected error
}
```

Output of `dafny /dprint:-`:
```
// Program.dfy

method Test() returns (r: int)
{
  var (x, y) := (123, 234);

  r := x;
}
Program.dfy(7,9): Error: ghost variables are allowed only in specification contexts
1 resolution/type errors detected in Program.dfy
```

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Fixes #1280

### Changes
- Add missing ghost modifier when printing VarDeclStmt
- Rename the property `IsAutoGhost` to `HasGhostModifier` and flip the boolean value. LMK if anyone has concerns.

### CODE REVIEW

**Comment by RustanLeino** [comment]
Nice changes!

Regarding testing, I suggest just adding one line to some test that currently uses `/dprint:-` or `/rprint:-`.

(A note about your more ambitious test suggestion: a small number of tests currently re-run Dafny on the `/dprint`'ed output: `LetExpr.dfy`, `SmallTests.dfy`, `Calculations.dfy`, `GhostPrint.dfy`, and `DirtyLoops.dfy`.)

File: Source/Dafny/DafnyMain.cs [hunk 1]
```diff
@@ -89,18 +89,12 @@ public class DafnyFile {
...
-              if (filename == "-") {
-                  tw = System.Console.Out;
-              } else {
-                  tw = new System.IO.StreamWriter(filename);
-              }
-              Printer pr = new Printer(tw, DafnyOptions.O.PrintMode);
-              pr.PrintProgram(program, afterResolver);
-          }
+      public static void MaybePrintProgram(Program program, string filename, bool afterResolver) {
+        if (filename == null) return;
```
Comment (RustanLeino) [inline Source/Dafny/DafnyMain.cs:93]: Please wrap curly braces around the then branch of the `if`.

File: Source/Dafny/DafnyAst.cs [hunk 2]
```diff
@@ -7469,13 +7469,13 @@ public class VarDeclPattern : Statement
   {
     public readonly CasePattern<LocalVariable> LHS;
     public readonly Expression RHS;
-    public bool IsAutoGhost;
+    public bool HasGhostModifier;
```
Comment (RustanLeino) [inline Source/Dafny/DafnyAst.cs:7472]: Nice.

**Review by RustanLeino — CHANGES_REQUESTED** [review]

**Comment by keyboardDrummer** [comment]
> Nice changes!
> 
> Regarding testing, I suggest just adding one line to some test that currently uses `/dprint:-` or `/rprint:-`.
> 
> (A note about your more ambitious test suggestion: a small number of tests currently re-run Dafny on the `/dprint`'ed output: `LetExpr.dfy`, `SmallTests.dfy`, `Calculations.dfy`, `GhostPrint.dfy`, and `DirtyLoops.dfy`.)

There's a test called PrettyPrinting which I added something to 😄

**Review by RustanLeino — APPROVED** [review]

**Pull request merged**